src/basic/socket-util.h

```diff
@@ -189,11 +189,16 @@ int flush_accept(int fd);
         })
 
 struct cmsghdr* cmsg_find(struct msghdr *mh, int level, int type, socklen_t length);
+void* cmsg_find_and_copy_data(struct msghdr *mh, int level, int type, void *buf, size_t buf_len);
 
 /* Type-safe, dereferencing version of cmsg_find() */
 #define CMSG_FIND_DATA(mh, level, type, ctype)                          \
         CMSG_TYPED_DATA(cmsg_find(mh, level, type, CMSG_LEN(sizeof(ctype))), ctype)
 
+/* Type-safe version of cmsg_find_and_copy_data() */
+#define CMSG_FIND_AND_COPY_DATA(mh, level, cmsg_type, type)             \
```

Comment:
Nitpick: It would be nice if `CMSG_FIND_DATA` and `CMSG_FIND_AND_COPY_DATA` used same names for the type parameters.